src/ProfessionalGraveyard.tsx

```diff
@@ -7,32 +7,80 @@ import { shortenAddress } from './Utilities';
 
 import Burned from './img/statistics-icons/Burned.png';
 
+import { Trait, Attribute } from './Types';
+
+import { Filter, Traits } from './Filter';
+
 export interface GraveyardProps {
     burntSlugs: BurntSlug[];
+    traitNameMap?: Map<string, Trait>;
+    attributes?: Attribute[];
 }
 
 export function ProfessionalGraveyard(props: GraveyardProps) {
     const {
         burntSlugs,
+        traitNameMap,
+        attributes,
     } = props;
 
-    const data = React.useMemo(() => {
+    const [showFilter, setShowFilter] = React.useState<boolean>();
+
+    const handleToggleFilter = () => {
+      setShowFilter((val) => !val);
+    }
+
+    const [dataFromFilter, setDataFromFilter] = React.useState<Traits>();
+
+    if (!dataFromFilter) {
```

Comment:
we shouldn't be running code like this on every render loop

it should either be done in the `useState` declaration of if we need to reset like this then use a useEffect